### PR TITLE
fix: unused table options

### LINKED ISSUE #1433: Table options can't be found in distributed mode
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Frontend

### What happened?

When implementing `SHOW CREATE` statement #1336 , I found that the table options can't be retrieved from table info, it's stored in the region on datanode.

After discussing with @v0y4g3r , @waynexia , maybe we can provide an RPC to get the table info from the table's first region.

### What operating system did you use?

Mac OS

### Relevant log output and stack trace

_No response_

### How can we reproduce the bug?

`show create table [table]` in distributed mode doesn't display the table options info.

### PR DESCRIPTION
**Pull request by NiwakaDev**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR:

* avoids saving unused table options.

* adds more tests for table options. (See https://github.com/GreptimeTeam/greptimedb/pull/2209#discussion_r1299605948)

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
close #1433, #2240

### CODE REVIEW

**Comment by MichaelScofield** [comment]
@NiwakaDev thx! there're conflicts, can you resolve them first?

**Comment by NiwakaDev** [comment]
I'm sorry for being late to resolve them.

File: tests-integration/src/tests/instance_test.rs [hunk 1]
```diff
@@ -151,6 +191,75 @@ PARTITION BY RANGE COLUMNS (ts) (
...
+|                  |   "c_bool" BOOLEAN NULL,                                             |
+|                  |   "c_date" DATE NULL,                                                |
+|                  |   "c_datetime" TIMESTAMP(0) NULL,                                    |
+|                  |                                                                      |
+|                  | )                                                                    |
+|                  |                                                                      |
+|                  | ENGINE=file                                                          |
+|                  | WITH(                                                                |
+|                  |   format = 'csv',                                                    |
+|                  |   location = '{location}', |
```
Comment (NiwakaDev) [inline tests-integration/src/tests/instance_test.rs:230]: I found that this is a flaky test because it is depending on the current dir name length....

Comment (MichaelScofield) [reply tests-integration/src/tests/instance_test.rs]: For sqlness tests, the output's format can be changed by some sqlness hints (the lines started with "--"). See the examples in "join_10_tables.result". 

For rust tests, you can simply strip the create table SQL out of the output recordbatch's vectors, and parse the SQL again to get the table options for comparation.

File: src/frontend/src/statement.rs [hunk 2]
```diff
@@ -326,3 +352,112 @@ fn extract_timestamp(map: &HashMap<String, String>, key: &str) -> Result<Option<
         })
         .transpose()
 }
+
+fn validate_table_options_keys(
```
[the remark this answers is not in the document]
Comment (NiwakaDev) [reply src/frontend/src/statement.rs:356]: >matches!(option, IMMUTABLE_TABLE_LOCATION_KEY | IMMUTABLE_TABLE_PATTERN_KEY | WRITE_BUFFER_SIZE_KEY ...)

We don't care the difference between s3 and the others?

[the remark this answers is not in the document]
Comment (MichaelScofield) [reply src/frontend/src/statement.rs:356]: No, as long as it belongs the pre-defined table options, we can say it's valid here. I guess you're concerning that what if the table options in the create table SQL mixed up, eg an Azure option in a bunch of S3 options. I think it's not the responsibility of the codes here. How to use and check the table options are things that actual implementors do.

**Comment by NiwakaDev** [comment]
@MichaelScofield 
PTAL. I guess that I addressed your review.

**Review by MichaelScofield — APPROVED** [review]

**Comment by MichaelScofield** [comment]
@NiwakaDev We are currently working on a major feature (region server), the develop branch is freezed. I have just changed the target of this PR to the working branch of that feature. Once the region server is done, we will rebase develop onto it.

File: tests/cases/distributed/show/show_create.sql [hunk 3]
```diff
@@ -26,3 +30,25 @@ create table table_without_partition (
...
+    PARTITION r0 VALUES LESS THAN (5),
+    PARTITION r1 VALUES LESS THAN (9),
+    PARTITION r2 VALUES LESS THAN (MAXVALUE),
+)
+ENGINE=mito
+WITH(
+  foo = 123,
+  ttl = '7d',
+  write_buffer_size = 1024
+);
```
Comment (evenyag) [inline tests/cases/distributed/show/show_create.sql:54]:
```suggestion
);

```

File: tests/cases/standalone/show/show_create.sql [hunk 4]
```diff
@@ -16,3 +16,20 @@ WITH(
...
+  ts TIMESTAMP NOT NULL DEFAULT current_timestamp(),
+  TIME INDEX (ts),
+  PRIMARY KEY (id, host)
+)
+ENGINE=mito
+WITH(
+  foo = 123,
+  ttl = '7d',
+  write_buffer_size = 1024
+);
```
Comment (evenyag) [inline tests/cases/standalone/show/show_create.sql:35]:
```suggestion
);

```

**Review by evenyag — APPROVED** [review]
LGTM. @killme2008 PTAL

**Pull request merged**